vm/src/stdlib/builtins.rs

```diff
@@ -338,7 +351,13 @@ mod builtins {
     }
 
     #[pyfunction]
-    fn hasattr(obj: PyObjectRef, attr: PyStrRef, vm: &VirtualMachine) -> PyResult<bool> {
+    fn hasattr(obj: PyObjectRef, attr: PyObjectRef, vm: &VirtualMachine) -> PyResult<bool> {
+        let attr = PyStrRef::try_from_object(vm, attr.clone()).map_err(|_e| {
+            vm.new_type_error(format!(
+                "attribute name must be string, not '{}'",
+                attr.class().name()
+            ))
+        })?;
```

Comment:
```suggestion
        let attr = attr.try_to_ref::<PyStr>(vm)?.map_err(|_e| {
            vm.new_type_error(format!(
                "attribute name must be string, not '{}'",
                attr.class().name()
            ))
        })?;
```
Another suggestion to remove unnecessary retain operation.

Reply:
I'm not really sure but the initial code I commited returns `&PyRef<PyStr>` but the suggestion seems to return `&Py<PyStr>`, which leads to an error.

Could I ask how to solve it? Thank you.

Reply:
Yes, they are different.
`PyStrRef::try_from_object` moves argument's value to its return value while `PyObject::try_to_ref` borrows `&Py<T>` from its argument.

Because `attr` is now a borrow, return part also need to be `Ok(vm.get_attribute_opt(obj, attr)?.is_some())`
without `&`.

If you still experience problem, i will look in it again. pasting error message will be helpful.

Reply:
Just changed the code based on your suggestion. I appreciate your help.